cfgmgr/intfmgr.cpp

```diff
@@ -39,8 +41,23 @@ bool IntfMgr::setIntfIp(const string &alias, const string &opCmd,
     {
         cmd << IP_CMD << " -6 address " << opCmd << " " << ipPrefixStr << " dev " << alias;
     }
-    int ret = swss::exec(cmd.str(), res);
-    return (ret == 0);
+    EXEC_WITH_ERROR_THROW(cmd.str(), res);
+}
+
+void IntfMgr::setIntfVrf(const string &alias, const string vrfName)
+{
+    stringstream cmd;
+    string res;
+
+    if (!vrfName.empty())
+    {
+        cmd << IP_CMD << " link set " << alias << " master " << vrfName;
```

Comment:
When the interface is enslaved in a vrf, the connected (subnet route) and receive (ip2me route) need to be removed from the old vrf. Is the expectation that SAI will do this automatically?

I don't see this PR taking care of that and I don't see #632  taking care of that either.

Please clarify whether it's been agreed that SAI will do this. If not, then either this PR or the one I mentioned above, need to take care of it, unless I've missed something.

Reply:
Seems https://github.com/Azure/sonic-swss/pull/632 won't be taking care of this. As such, this PR will need to address the issue.

Reply:
@nikos-github , for the current use-case, the interface configuration (in ConfigDB) will have the interface with the VRF name. That means, when it is created first time, it will be enslaved to this VRF. My understanding is in phase 1, we are not planning for dynamically changing the interface from one VRF to another. When we have the full VRF implementation as part of https://github.com/Azure/SONiC/pull/242, this shall be taken care.